[Native Image] Cannot use tool provider javac using Graal

### Describe the Issue

I am experimenting with using the `ToolProvider` in a build tool and using native-image. This does not seem to work as the resources for compiler messages are stripped away. I tried to identify the missing resource, but with no success. Is there a known way to use `jdk.compiler` within a native image? I am using the latest version of Graal.

```
Caused by: java.lang.InternalError: Cannot find requested resource bundle for locale en_US
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.getBundles(JavacMessages.java:145)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.setCurrentLocale(JavacMessages.java:80)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.<init>(JavacMessages.java:114)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.<init>(JavacMessages.java:90)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.instance(JavacMessages.java:58)
	at jdk.compiler@25/com.sun.tools.javac.util.JCDiagnostic$Factory.<init>(JCDiagnostic.java:74)
	at jdk.compiler@25/com.sun.tools.javac.util.JCDiagnostic$Factory.instance(JCDiagnostic.java:63)
	at jdk.compiler@25/com.sun.tools.javac.util.Log.<init>(Log.java:334)
	at jdk.compiler@25/com.sun.tools.javac.util.Log.<init>(Log.java:269)
	at jdk.compiler@25/com.sun.tools.javac.util.Log.instance(Log.java:248)
	at jdk.compiler@25/com.sun.tools.javac.main.Main.compile(Main.java:205)
	at jdk.compiler@25/com.sun.tools.javac.main.Main.compile(Main.java:178)
	at jdk.compiler@25/com.sun.tools.javac.main.JavacToolProvider.run(JavacToolProvider.java:54)
	at build.buildbuddy.step.ProcessHandler$OfTool.execute(ProcessHandler.java:47)
	at build.buildbuddy.step.ProcessBuildStep.lambda$apply$1(ProcessBuildStep.java:50)
	... 4 more
Caused by: java.util.MissingResourceException: Can't find bundle for base name com.sun.tools.javac.resources.compiler, locale en_US
	at java.base@25/java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:2012)
	at java.base@25/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1664)
	at java.base@25/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:122)
	at java.base@25/java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1529)
	at java.base@25/java.util.ResourceBundle.getBundle(ResourceBundle.java:921)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.lambda$add$0(JavacMessages.java:123)
	at jdk.compiler@25/com.sun.tools.javac.util.JavacMessages.getBundles(JavacMessages.java:142)
	... 18 more

```

### Using the latest version of GraalVM can resolve many issues.

- [x] I tried with the [latest version](https://www.graalvm.org/downloads/) of GraalVM.

### GraalVM Version

java version "25" 2025-09-16 LTS
Java(TM) SE Runtime Environment Oracle GraalVM 25-dev+6.1 (build 25+6-LTS-jvmci-b01)
Java HotSpot(TM) 64-Bit Server VM Oracle GraalVM 25-dev+6.1 (build 25+6-LTS-jvmci-b01, mixed mode, sharing)


### Operating System and Version

Ubuntu 24.04

### Troubleshooting Confirmation

- [x] I tried the suggestions in the [troubleshooting guide](https://www.graalvm.org/latest/reference-manual/native-image/guides/troubleshoot-run-time-errors/).

### Run Command

native-image -H:+UnlockExperimentalVMOptions -H:IncludeResources=".*" build.Manual

### Expected Behavior

Creates build tool that can run *javac*

### Actual Behavior

The mentioned exception occurs.

### Steps to Reproduce

git clone https://github.com/raphw/build-buddy
cd build-buddy
javac build/Manual.java
native-image build.Manual
./build.manual

### Additional Context

_No response_

### Run-Time Log Output and Error Messages

_No response_

--- a/build/Manual.java
+++ b/build/Manual.java
@@ -15,6 +15,11 @@ import java.util.Map;
 public class Manual {
 
     public static void main(String[] args) throws IOException {
+        var javaHomeEnv = System.getenv("JAVA_HOME");
+        if (javaHomeEnv == null) {
+            throw new IllegalStateException("JAVA_HOME environment variable is not set; please set it to compile Java");
+        }
+        System.setProperty("java.home", javaHomeEnv);
         MavenRepository mavenRepository = new MavenDefaultRepository();
         Map<String, Repository> repositories = Map.of("maven", mavenRepository);
         Map<String, Resolver> resolvers = Map.of("maven", new MavenPomResolver());

--- a/config/reachability-metadata.json
+++ b/config/reachability-metadata.json
@@ -1,0 +1,19 @@
+{
+  "includes": [
+    { "pattern": ".*" }
+  ],
+  "bundles": [
+    {
+      "name": "com.sun.tools.javac.resources.compiler",
+      "locales": ["en"]
+    },
+    {
+      "name": "sun.tools.jar.resources.jar",
+      "locales": ["en"]
+    },
+    {
+      "name": "com.sun.tools.javac.resources.javac",
+      "locales": ["en"]
+    }
+  ]
+}

--- a/sources/build/buildbuddy/step/Java.java
+++ b/sources/build/buildbuddy/step/Java.java
@@ -1,8 +1,5 @@
 package build.buildbuddy.step;
 
-import build.buildbuddy.BuildStepArgument;
-import build.buildbuddy.BuildStepContext;
-
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.FileVisitResult;
@@ -20,6 +17,9 @@ import java.util.function.Function;
 import java.util.jar.JarFile;
 import java.util.stream.Stream;
 import java.util.zip.ZipFile;
+
+import build.buildbuddy.BuildStepArgument;
+import build.buildbuddy.BuildStepContext;
 
 public abstract class Java extends ProcessBuildStep {
 
@@ -122,6 +122,7 @@ public abstract class Java extends ProcessBuildStep {
             }
         }
         List<String> prefixes = new ArrayList<>();
+        prefixes.add("--enable-preview");
         if (!classPath.isEmpty()) {
             prefixes.add("-classpath");
             prefixes.add(String.join(File.pathSeparator, classPath));

--- a/sources/build/buildbuddy/step/Javac.java
+++ b/sources/build/buildbuddy/step/Javac.java
@@ -1,8 +1,5 @@
 package build.buildbuddy.step;
 
-import build.buildbuddy.BuildStepArgument;
-import build.buildbuddy.BuildStepContext;
-
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.FileVisitResult;
@@ -17,6 +14,9 @@ import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.Executor;
 import java.util.function.Function;
+
+import build.buildbuddy.BuildStepArgument;
+import build.buildbuddy.BuildStepContext;
 
 public class Javac extends ProcessBuildStep {
 
@@ -39,6 +39,7 @@ public class Javac extends ProcessBuildStep {
             throws IOException {
         Path target = Files.createDirectory(context.next().resolve(CLASSES));
         List<String> files = new ArrayList<>(), path = new ArrayList<>(), commands = new ArrayList<>(List.of(
+                "--enable-preview",
                 "--release", Integer.toString(Runtime.version().version().getFirst()),
                 "-d", target.toString()));
         for (BuildStepArgument argument : arguments.values()) {

--- a/sources/build/buildbuddy/step/Tests.java
+++ b/sources/build/buildbuddy/step/Tests.java
@@ -1,8 +1,5 @@
 package build.buildbuddy.step;
 
-import build.buildbuddy.BuildStepArgument;
-import build.buildbuddy.BuildStepContext;
-
 import java.io.IOException;
 import java.nio.file.FileVisitResult;
 import java.nio.file.Files;
@@ -19,6 +16,9 @@ import java.util.function.Function;
 import java.util.function.Predicate;
 import java.util.regex.Pattern;
 import java.util.stream.Stream;
+
+import build.buildbuddy.BuildStepArgument;
+import build.buildbuddy.BuildStepContext;
 
 public class Tests extends Java {
 
@@ -63,6 +63,7 @@ public class Tests extends Java {
                 .of(() -> arguments.values().stream().map(BuildStepArgument::folder).iterator())
                 .orElseThrow(() -> new IllegalArgumentException("No test engine found")) : this.engine;
         List<String> commands = new ArrayList<>();
+        commands.add("--enable-preview");
         if (modular && engine.module != null) {
             commands.add("--add-modules");
             commands.add("ALL-MODULE-PATH");

